crates/forge/bin/cmd/test/mod.rs

```diff
@@ -132,6 +138,61 @@ impl TestArgs {
         self.execute_tests().await
     }
 
+    pub fn get_sources_to_compile(
+        &self,
+        config: &Config,
+        filter: &ProjectPathsAwareFilter,
+    ) -> Result<BTreeSet<PathBuf>> {
+        let mut project = config.create_project(true, true)?;
+        project.solc_config.settings.output_selection.0 = BTreeMap::from([(
+            "*".to_string(),
+            BTreeMap::from([("*".to_string(), vec!["abi".to_string()])]),
+        )]);
```

Comment:
I think there's an easier way to do this, if not we should make this easier in foundry-compilers